Caboodle/Permissions/Permissions.android.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Android;
+using Android.Content.PM;
+using Android.Support.V4.App;
+using Android.Support.V4.Content;
+
+namespace Microsoft.Caboodle
+{
+    internal static partial class Permissions
+    {
+        static TaskCompletionSource<PermissionStatus> tcs;
```

Comment:
Don 't love this being a static... It really means we can only ever have one request going at a time, which should be the case, but who knows...

I'm thinking maybe a `Dictionary<int, TaskCompletionSource<PermissionStatus>>` is a better idea (where `int` is the permission request code that we keep incrementing with each call).

Reply:
yeah, that seems like a better idea